backend/src/main/java/org/cryptomator/hub/license/LicenseHolder.java

```diff
@@ -75,6 +85,45 @@ public void set(String token) throws JWTVerificationException {
 		settings.persist();
 	}
 
+	/**
+	 * Attempts to refresh the Hub licence every day at 01:00 UTC if claim refreshURL is present.
+	 */
+	@Scheduled(cron = "0 1 * * * ?", timeZone = "UTC")
+	void refreshLicenseScheduler() throws InterruptedException {
+		if (license != null) {
+			var refreshUrl = licenseValidator.refreshUrl(license.getToken());
+			if (refreshUrl.isPresent()) {
+				var client = HttpClient.newBuilder().followRedirects(HttpClient.Redirect.NORMAL).build();
+				refreshLicense(refreshUrl.get(), license.getToken(), client);
+			}
+		}
+	}
+
+	//visible for testing
+	void refreshLicense(String refreshUrl, String license, HttpClient client) throws InterruptedException {
+		var parameters = new HashMap<String, String>();
+		parameters.put("token", license);
+		var body = parameters.entrySet() //
+				.stream() //
+				.map(e -> e.getKey() + "=" + URLEncoder.encode(e.getValue(), StandardCharsets.UTF_8)) //
```

Comment:
I am not sure if URLEncoder does the right thing here. If I am not mistaken, it encodes stuff like ` ` to `%20` but it *does not* encode characters that are allowed in URLs, such as `&` and `=`.

What we really need is an encoder that encodes every character which is not allowed in *query parameters*.

Add a unit test which tests this method with `license = "foo&bar=baz"`.

Reply:
AFAIK it does exactly what we want, from https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/net/URLEncoder.html

> When encoding a String, the following rules apply:
> * The alphanumeric characters "a" through "z", "A" through "Z" and "0" through "9" remain the same.
> * The special characters ".", "-", "*", and "_" remain the same.
> * The space character "   " is converted into a plus sign "+".
> * All other characters are unsafe and are first converted into one or more bytes using some encoding scheme. > Then each byte is represented by the 3-character string "%xy", where xy is the two-digit hexadecimal representation of the byte. The recommended encoding scheme to use is UTF-8. However, for compatibility reasons, if an encoding is not specified, then the default encoding of the platform is used. 
> 
> For example using UTF-8 as the encoding scheme the string "The string ü@foo-bar" would get converted to "The+string+%C3%BC%40foo-bar" because in UTF-8 the character ü is encoded as two bytes C3 (hex) and BC (hex), and the character @ is encoded as one byte 40 (hex).

Added a unit test in [174c7aa](https://github.com/cryptomator/hub/pull/246/commits/174c7aad99b66f5373fc5c46b696b96ec41ad1d2)

Reply:
Nevermind, URLEncoder seems perfectly fitting for this job. Such a unit test would be a good addition anyway :)

Reply:
There is one, I changed the first test scenario to verify it: https://github.com/cryptomator/hub/blob/refresh-license/backend/src/test/java/org/cryptomator/hub/license/LicenseHolderTest.java#L221-L256

There I'm requesting a license for `token&foo=bar` which results in the encoded request body `token=token%26foo%3Dbar`.